### PR TITLE
Add localization to user config page (#3845)

### LINKED ISSUE #3845: Make user config page translatable
All text in the Home Assistant UI should be translated.

The [User config panel](https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/panels/config/users/ha-config-users.js) is not translated yet.
- Create new keys in `src/translations/en.json` with the text that is currently hard coded in the source.
Try to follow the same logic that is present, and give every section in Home Assistant it's own object.
If you would place the text in
```json
  "ui": {
    "panel": {
      "example": {
        "text": "Here goes the text",
      },
    },
  },
```
your key would be: `"ui.panel.example.text`

Polymer Elements need a mixin: `LocalizeMixin(PolymerElement)`, Lit Elements work without a mixin. 

- Replace the text in the code with:
  - Lit element: `${this.hass.localize("ui.panel.example.text")}`
  - Polymer: `[[localize('ui.panel.history.showing_entries')]]`

_(This is part of https://github.com/home-assistant/home-assistant-polymer/issues/3429)_

### PR DESCRIPTION
**Pull request by svendroid**
Closes #3845

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @svendroid, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant-polymer#3869).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: src/panels/config/users/ha-user-editor.ts [hunk 1]
```diff
@@ -154,13 +171,23 @@ class HaUserEditor extends LitElement {
...
     }
   }
 
   private async _deleteUser(ev): Promise<void> {
-    if (!confirm(`Are you sure you want to delete ${this._name}`)) {
+    if (
+      !confirm(
+        `${this.hass!.localize(
+          "ui.panel.config.users.editor.confirm_user_deletion"
+        )} ${this._name}`
```
Comment (bramkragten) [inline src/panels/config/users/ha-user-editor.ts]: Lets give `this._name` as a variable to localize:
```js
${this.hass!.localize("ui.panel.config.users.editor.confirm_user_deletion", "name", this._name)}
```
and add the variable name to the translation: `{name}`

**Review by bramkragten — APPROVED** [review]
Looks good thanks! 👍

File: src/panels/config/users/ha-user-editor.ts (hunk 1 above)
Comment (svendroid) [reply src/panels/config/users/ha-user-editor.ts]: Applied it in latest change

**Review by bramkragten — APPROVED** [review]
Great! 🐬

**Pull request merged**